src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -126,6 +126,7 @@ def _validate_result(self, result: Optional[str]) -> Optional[int]:
 
         if not self._is_valid_timestamp(timestamp):
             metrics.incr(_INDEXER_CACHE_STALE_KEYS_METRIC)
+            return None
```

Comment:
Since this doesn't delete the key/value pair from the cache, can we add a test that checks that the Postgres `bulk_record` refreshes the timestamp for the cache key?

Reply:
Add to the test, but I did not use postgres indexer in it, because we don't really have any tests that involves both postgres indexer and the caching indexer.
The new section of the test call `set_many` on the stale keys and verify that is has the proper (< 2 hr old) timestamps